strawberryfields/api_client.py

```diff
@@ -0,0 +1,101 @@
+from urllib.parse import urljoin
+import requests
+
+
+class APIClient:
+    ALLOWED_BASE_URLS = [
+        'localhost',
+    ]
+    DEFAULT_BASE_URL = 'localhost/'
+    CONFIGURATION_PATH = ''
+
+    def __init__(self, *args, **kwargs):
+        # TODO: Load username, password, or authentication token from
+        # configuration file
+
+        self.USE_SSL = kwargs.get('use_ssl', True)
+        self.AUTHENTICATION_TOKEN = kwargs.get('authentication_token', '')
+        self.HEADERS = {}
+
+        if 'headers' in kwargs:
+            self.HEADERS.update(kwargs['headers'])
+
+        if 'base_url' in kwargs:
+            base_url = kwargs['base_url']
+            if base_url in self.ALLOWED_BASE_URLS:
+                self.BASE_URL = base_url
+            else:
+                raise ValueError('base_url parameter not in allowed list')
+        else:
+            self.BASE_URL = self.DEFAULT_BASE_URL
```

Comment:
here, you could use
```python
self.BASE_URL = kwargs.get('base_url', self.DEFAULT_BASE_URL)

if self.base_url not in self.ALLOWED_BASE_URLS:
    raise ValueError('base_url parameter not in allowed list')
```

to avoid the outer if statement. But I guess we'll be moving the base URL to the configuration file, so this is not too important to change now.

Reply:
Makes sense, I originally used `kwargs.get` but switched to this as part of a code change and didn't refactor after changing the logic.